test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala

```diff
@@ -435,6 +435,98 @@ class RegressionTest {
     assertEquals((Nil, 10), result)
   }
 
+  @Test def eqEqJLDouble(): Unit = {
+    // Taken from run/sd329.scala in scala/scala
+
+    val v = Platform.scalaVersion
+    assumeFalse("Bug scala-dev#329, #2799 on this version of Scala",
+        v.startsWith("2.10.") || v.startsWith("2.11.") ||
+        v == "2.12.0" || v == "2.12.1")
```

Comment:
Shouldn't we also add tests for the other case? I.e. when this assume is true? Seems simple enough and guards us against not being really bug compatible.